### ISSUE TITLE
Update progress to any arbitrary step

### ISSUE THREAD

**Issue by kelly-sovacool** [HenrikBengtsson/progressr#110]
The progress package allows you to jump the progress bar to any arbitrary step by supplying a fraction to `update()`:

```r
p <- progress::progress_bar$new(total = 5, clear = FALSE)
for (i in 1:3) {
    p$tick()
    Sys.sleep(0.5)
}
#> [=======================================>---------------------------]  60%
p$update(1)
#> [===================================================================] 100%
```

This is very useful when using a progress bar without an iterator, or when you miscalculate the total number of steps required when creating the progress bar such as when the total number of steps is not known until after initialization.

It would be nice to have a similar feature for `progressr`. I imagine it could be an optional argument to the progressor function, perhaps something like:

```r
p <- progressr::progressor(steps = 5)
for (i in 1:3) {
    p()
    Sys.sleep(0.5)
}
#> [=======================================>---------------------------]  60%
p(step = 5)
#> [===================================================================] 100%
```

**Comment by HenrikBengtsson** [comment]
Hi, thanks for the suggestion.  I think there was an idea in the early days to support this, because I had already prepared a `step` argument;

https://github.com/HenrikBengtsson/progressr/blob/61355f962811233fb527c2afe64357d2683937d6/R/progression.R#L68-L80

but it was never used.  One reason for not being used might have been that I was concerned about not being able to control the order of when progress updates arrive to the handler, e.g. when running in parallel.  Being able to only specify relative updates (=`amount`) such problems are completely avoided. 

Anyway, and although the above is still a concern, the update required was tiny, so I went ahead and implemented this (commit 9bb85f).  However, I marked it as a beta feature, in case it turns out to be a bad idea and we have to unroll it again. I'm hopeful that won't be needed.  But please use with care after reading the warning in the NEWS file.

**Comment by kelly-sovacool** [comment]
Thanks @HenrikBengtsson! So would you recommend _not_ using this experimental feature when running tasks in parallel?

**Comment by HenrikBengtsson** [comment]
It's ok when you're in full control of the order, as in you example.  An example where you're likely to end up with incorrect total progress reported is cases like:

```r
my_fcn <- function(X) {
  p <- progressr::progressor(along = X)
  y <- future_lapply(X, FUN = function(x) {
    p()
    if (x > 5) p(step = 8)
    slow_processing(x)
  })
  p()
  y
}
```

In that case, you never know in what order the "step=8" progression condition is signaled relative to the other "amount=1.0" progress conditions.  That depends on which parallel backend is used and relative processing time for each element in `X`.  So, they might arrive to the handler as:

```r
p()
p()
p()
p(step = 8)
p()
...
```

or

```r
p(step = 8)
p()
p()
p()
p()
...
```

In contrast, if you use it as in:

```r
my_fcn <- function(X) {
  p <- progressr::progressor(along = X)
  y <- future_lapply(X, FUN = function(x) {
    p()
    slow_processing(x)
  })
  p(step = 8)
  # Some more progress updates
  p()
  y
}
```

you should be fine because you know exactly what comes before and after the "step=8" progression.

Does this make sense?

**Comment by kelly-sovacool** [comment]
Yes, your code examples are very helpful. Thanks @HenrikBengtsson!

**Comment by HenrikBengtsson** [comment]
FYI, **progressr** 0.8.0 fixing this just reached CRAN.
